pkg/asset/cluster/tfvars.go

```diff
@@ -436,13 +437,22 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		// Search the project for a dns zone with the specified base domain.
 		publicZoneName := ""
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			publicZone, err := gcpconfig.GetPublicZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain)
+			publicZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain, true)
 			if err != nil {
 				return errors.Wrapf(err, "failed to get GCP public zone")
 			}
 			publicZoneName = publicZone.Name
 		}
 
+		privateZoneName := ""
+		privateZone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.ClusterDomain(), false)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get GCP private zone")
+		}
+		if privateZone != nil {
+			privateZoneName = privateZone.Name
+		}
+
```

Comment:
In order to only enable this functionality with GCP XPN, I think we need to also wrap this code block with

```go
// Using an existing hosted zone is only supported in GCP XPN scenarios
if installconfig.Config.GCP.NetworkProjectID !== "" {
```